aws.go

```diff
@@ -109,29 +109,43 @@ func scrape(collector *cwCollector, ch chan<- prometheus.Metric) {
 
 		
 		// Get all the metric to select the ones who'll match the regex
+		now = time.Now()
 		result, err := svc.ListMetrics(&cloudwatch.ListMetricsInput{
 			MetricName: aws.String(metric.ConfMetric.Name),
 			Namespace:  aws.String(metric.ConfMetric.Namespace),
 		})
 		nextToken:=result.NextToken
 		metrics:=result.Metrics
 		totalRequests.Inc()
+		collector.ScrapeDurationHistogram.Observe(time.Since(now).Seconds())
 
 		if err != nil {
+			totalErrors.Inc()
+			collector.ErroneousRequests.Inc()
 			fmt.Println(err)
 			continue
 		}
+		collector.SuccessfulRequests.Inc()
 
 		for nextToken!=nil {
+			now = time.Now()
 			result, err := svc.ListMetrics(&cloudwatch.ListMetricsInput{
 				MetricName: aws.String(metric.ConfMetric.Name),
 				Namespace:  aws.String(metric.ConfMetric.Namespace),
 				NextToken: nextToken,
-			})		
+			})	
+
+			totalRequests.Inc()
+			collector.ScrapeDurationHistogram.Observe(time.Since(now).Seconds())
+	
 			if err != nil {
+				totalErrors.Inc()
+				collector.ErroneousRequests.Inc()
 				fmt.Println(err)
 				continue
 			}
+			collector.SuccessfulRequests.Inc()
```

Comment:
Now I see why errors increment goes in 2s, this will increment for errors too after err condition

Reply:
`continue` statement on L133 jumps out to next iteration of loop on error.